### ISSUE TITLE
@InheritReverseConfiguration cannot directly inherit from prototype methods

### ISSUE THREAD

**Issue by sjaakd** [mapstruct/mapstruct#1065]
I have the simple use case: 
1. SharedConfig
2. Reuse SharedConfig to separate from-to, to-from into 2 separate classes, define the mappings once.

It does not work..

Applied to our unit test: `org.mapstruct.ap.test.inheritfromconfig.InheritFromConfigTest`, this is the reproducer:

```java
@Mapper(
    uses = NotToBeUsedMapper.class,
    config = AutoInheritedConfig.class,
    mappingInheritanceStrategy = MappingInheritanceStrategy.EXPLICIT
)
public abstract class CarMapperReverseWithExplicitInheritance {
    public static final CarMapperReverseWithExplicitInheritance INSTANCE =
        Mappers.getMapper( CarMapperReverseWithExplicitInheritance.class );

    @InheritInverseConfiguration(name = "baseDtoToEntity")
    public abstract CarDto toCarDto(CarEntity entity);

}
```

Note: a similar method is there in `CarMapperWithExplicitInheritance`, but there the reverse mapping runs via a local present method.

**Comment by sjaakd** [comment]
@agudian , @gunnarmorling : do you guys remember why I can't inherit inverse mappings from a `SharedConfig`. The use case sketched above seems to justify such mappings. In my head, these `@InheritConfiguration` and `@InheritInverseConfiguration` were more or less symmetric. Yet the code seems to be designed with only `@InheritConfiguration` in mind when it comes to prototype methods..

**Comment by agudian** [comment]
I don't think there was a specific reason for not implementing it -- I just didn't think of it, I guess. At least I can't remember anything else... 😁 

Perhaps it can be fixed when the stuff get's refactored for making them applicable in the forged methods?

**Comment by sjaakd** [comment]
Good to know.. I've got a PR in the making.. However UT's start to fail (because they refer to local reverse methods)..

**Comment by gunnarmorling** [comment]
Done.

**Comment by filiphr** [comment]
I have changed the milestone of this one, as it was fixed for `1.2.0.Beta2` actually. I have updated the migration notes as well.
